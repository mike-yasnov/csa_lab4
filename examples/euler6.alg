// Эйлер задача 6: разность квадрата суммы и суммы квадратов первых 100 натуральных чисел
// Квадрат суммы: (1 + 2 + ... + 100)^2
// Сумма квадратов: 1^2 + 2^2 + ... + 100^2

function main() {
    var n = 100;
    var sum = 0;
    var sum_squares = 0;
    
    // Вычисляем сумму и сумму квадратов
    for (var i = 1; i <= n; i = i + 1) {
        sum = sum + i;
        sum_squares = sum_squares + i * i;
    }
    
    // Квадрат суммы
    var square_of_sum = sum * sum;
    
    // Разность
    var difference = square_of_sum - sum_squares;
    
    print("Квадрат суммы: ");
    print(square_of_sum);
    print("\nСумма квадратов: ");
    print(sum_squares);
    print("\nРазность: ");
    print(difference);
}

main(); 