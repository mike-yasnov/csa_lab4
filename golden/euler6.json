{
  "return_code": 0,
  "stdout": "Загрузка программы: /tmp/tempfile/program.bin\nЗагружено 71 инструкций\nЗагрузка данных: /tmp/tempfile/program_data.bin\nЗагружено 325 байт данных\nЗапуск выполнения (максимум 1000000 тактов)...\n\n=== РЕЗУЛЬТАТЫ ВЫПОЛНЕНИЯ ===\nСостояние: halted\nВыполнено инструкций: 3\nЗатрачено тактов: 8\nФинальный PC: 2\n\nВЫВОД ПРОГРАММЫ:\nEuler [208][183][208][176][208][180][208][176][209][135][208][176] 6:\n\nПрограмма завершена успешно.\n",
  "stderr": "",
  "exec_log": "Cycle 000002: PC=0000 00(000000) Stack[0] TOS=empty\nCycle 000007: PC=0001 61(000001) Stack[1] TOS=0\nCycle 000008: PC=0002 01(000000) Stack[0] TOS=empty",
  "debug_listing": "INSTRUCTION MEMORY:\nAddress - Hex Code - Mnemonic\n----------------------------------------\n0000 - 00000000 - PUSH\n0002 - 61010000 - OUT 1\n0004 - 01000000 - POP\n0005 - 00160000 - PUSH 22\n0007 - 61010000 - OUT 1\n0009 - 01000000 - POP\n000A - 00640000 - PUSH 100\n000C - 00800000 - PUSH 128\n000E - 51000000 - STORE\n000F - 00800000 - PUSH 128\n0011 - 50000000 - LOAD\n0012 - 00800000 - PUSH 128\n0014 - 50000000 - LOAD\n0015 - 00010000 - PUSH 1\n0017 - 10000000 - ADD\n0018 - 12000000 - MUL\n0019 - 00020000 - PUSH 2\n001B - 13000000 - DIV\n001C - 00840000 - PUSH 132\n001E - 51000000 - STORE\n001F - 00880000 - PUSH 136\n0021 - 61010000 - OUT 1\n0023 - 01000000 - POP\n0024 - 00840000 - PUSH 132\n0026 - 50000000 - LOAD\n0027 - 00840000 - PUSH 132\n0029 - 50000000 - LOAD\n002A - 12000000 - MUL\n002B - 00B50000 - PUSH 181\n002D - 51000000 - STORE\n002E - 00B90000 - PUSH 185\n0030 - 61010000 - OUT 1\n0032 - 01000000 - POP\n0033 - 00800000 - PUSH 128\n0035 - 50000000 - LOAD\n0036 - 00800000 - PUSH 128\n0038 - 50000000 - LOAD\n0039 - 00010000 - PUSH 1\n003B - 10000000 - ADD\n003C - 12000000 - MUL\n003D - 00020000 - PUSH 2\n003F - 00800000 - PUSH 128\n0041 - 50000000 - LOAD\n0042 - 12000000 - MUL\n0043 - 00010000 - PUSH 1\n0045 - 10000000 - ADD\n0046 - 12000000 - MUL\n0047 - 00060000 - PUSH 6\n0049 - 13000000 - DIV\n004A - 00DD0000 - PUSH 221\n004C - 51000000 - STORE\n004D - 00E10000 - PUSH 225\n004F - 61010000 - OUT 1\n0051 - 01000000 - POP\n0052 - 00B50000 - PUSH 181\n0054 - 50000000 - LOAD\n0055 - 00DD0000 - PUSH 221\n0057 - 50000000 - LOAD\n0058 - 11000000 - SUB\n0059 - 00070100 - PUSH 263\n005B - 51000000 - STORE\n005C - 000B0100 - PUSH 267\n005E - 61010000 - OUT 1\n0060 - 01000000 - POP\n0061 - 00260100 - PUSH 294\n0063 - 61010000 - OUT 1\n0065 - 01000000 - POP\n0066 - 44000000 - RET\n0067 - 43000000 - CALL\n0069 - 01000000 - POP\n006A - 70000000 - HALT\n\nDATA MEMORY:\nAddress - Hex Dump - ASCII\n----------------------------------------\n0000 - 45 75 6C 65 72 20 D0 B7 D0 B0 D0 B4 D0 B0 D1 87  - Euler ..........\n0010 - D0 B0 20 36 3A 00 D0 A0 D0 B0 D0 B7 D0 BD D0 BE  - .. 6:...........\n0020 - D1 81 D1 82 D1 8C 20 D0 BA D0 B2 D0 B0 D0 B4 D1  - ...... .........\n0030 - 80 D0 B0 D1 82 D0 B0 20 D1 81 D1 83 D0 BC D0 BC  - ....... ........\n0040 - D1 8B 20 D0 B8 20 D1 81 D1 83 D0 BC D0 BC D1 8B  - .. .. ..........\n0050 - 20 D0 BA D0 B2 D0 B0 D0 B4 D1 80 D0 B0 D1 82 D0  -  ...............\n0060 - BE D0 B2 20 D0 BF D0 B5 D1 80 D0 B2 D1 8B D1 85  - ... ............\n0070 - 20 31 30 30 20 D1 87 D0 B8 D1 81 D0 B5 D0 BB 00  -  100 ...........\n0080 - 00 00 00 00 00 00 00 00 D0 A1 D1 83 D0 BC D0 BC  - ................\n0090 - D0 B0 20 D0 BF D0 B5 D1 80 D0 B2 D1 8B D1 85 20  - .. ............ \n00A0 - 31 30 30 20 D1 87 D0 B8 D1 81 D0 B5 D0 BB 3A 20  - 100 ..........: \n00B0 - 35 30 35 30 00 00 00 00 00 D0 9A D0 B2 D0 B0 D0  - 5050............\n00C0 - B4 D1 80 D0 B0 D1 82 20 D1 81 D1 83 D0 BC D0 BC  - ....... ........\n00D0 - D1 8B 3A 20 32 35 35 30 32 35 30 30 00 00 00 00  - ..: 25502500....\n00E0 - 00 D0 A1 D1 83 D0 BC D0 BC D0 B0 20 D0 BA D0 B2  - ........... ....\n00F0 - D0 B0 D0 B4 D1 80 D0 B0 D1 82 D0 BE D0 B2 3A 20  - ..............: \n0100 - 33 33 38 33 35 30 00 00 00 00 00 D0 A0 D0 B0 D0  - 338350..........\n0110 - B7 D0 BD D0 BE D1 81 D1 82 D1 8C 3A 20 32 35 31  - ...........: 251\n0120 - 36 34 31 35 30 00 D0 A0 D0 B5 D1 88 D0 B5 D0 BD  - 64150...........\n0130 - D0 B8 D0 B5 20 D0 BD D0 B0 D0 B9 D0 B4 D0 B5 D0  - .... ...........\n0140 - BD D0 BE 21 00                                   - ...!.\n"
}